packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -8,26 +8,95 @@ import {
     IFluidObject,
     IFluidRouter,
     IRequest,
+    IResponse,
 } from "@fluidframework/core-interfaces";
 import {
     IFluidDataStoreFactory,
     IFluidDataStoreRegistry,
     IProvideFluidDataStoreRegistry,
 } from "@fluidframework/runtime-definitions";
 
+interface IResponseException extends Error {
+    errorFromRequestFluidObject: true;
+    message: string;
+    code: number;
+    stack: string;
+}
+
+export function getStack() {
+    const err = new Error();
+    if (err.stack !== undefined) {
+        return err.stack;
+    }
+    try {
+        throw err;
+    } catch (err2) {
+        return (err2 as Error).stack;
+    }
+}
+
+export function exceptionToResponse(err: any): IResponse {
+    const status = 500;
+    // eslint-disable-next-line no-null/no-null
+    if (err !== null && typeof err === "object" && err.errorFromRequestFluidObject === true) {
+        const responseErr: IResponseException = err;
+        return {
+            mimeType: "text/plain",
+            status,
+            value: responseErr.message,
+            stack: responseErr.stack ?? getStack(),
+        };
+    }
+    return {
+        mimeType: "text/plain",
+        status,
+        value : `${err}`,
+        stack: getStack(),
+    };
+}
+
+export function responseToException(response: IResponse, request: IRequest) {
+    // add one more 'frame' to message
+    const message = `${response.value}\nNot found: ${request.url}`;
```

Comment:
Is there PII concern for telemetry to include the URL?

Reply:
There is definitely a concern, once we have 3rd party code in container. @markfields is looking into it, but I believe that the way we will solve it is through
1. annotating all properties on error objects in terms of PII or other type of sensitive info 
2. Have filtering code at host level, that depending on mark up (or lack of) will strip or leave such info.
This case is no different than data store ID / DDS id - it's safe to collect for all environments except when we have 3rd party code.

I believe right thing to do here is to open an issue for this particular case to track as part of the work that Mark is doing.

Note that not collecting this info unconditionally will make it impossible to investigate bugs like [this one](https://office.visualstudio.com/OC/_workitems/edit/4771313?src=WorkItemMention&src-action=artifact_link)

Reply:
I've opened https://github.com/microsoft/FluidFramework/issues/5326 to follow up here.